Opcodes/vbap_zak.c

```diff
@@ -27,580 +27,4 @@
-    }
-    /* Now read from the array in za space and write to the output. */
-    p->out_array     = zastart + (indx * CS_KSMPS);/* outputs */
-    csound->AuxAlloc(csound, p->n*sizeof(MYFLT)*4, &p->auxch);
-    p->curr_gains    = (MYFLT*)p->auxch.auxp;
-    p->beg_gains     = p->curr_gains + p->n;
-    p->end_gains     = p->beg_gains + p->n;
-    p->updated_gains = p->end_gains + p->n;
-    /* reading in loudspeaker info */
-    ls_table = (MYFLT*) (csound->QueryGlobalVariableNoCheck(csound,
-                                                        "vbap_ls_table_0"));
-    p->dim           = (int32_t) ls_table[0];
-    p->ls_am         = (int32_t) ls_table[1];
-    p->ls_set_am     = (int32_t) ls_table[2];
-    ptr              = &(ls_table[3]);
-    csound->AuxAlloc(csound, p->ls_set_am * sizeof (LS_SET), &p->aux);
-    if (UNLIKELY(p->aux.auxp == NULL)) {
-      return csound->InitError(csound, "%s", Str("could not allocate memory"));
-    }
-    p->ls_sets = (LS_SET*) p->aux.auxp;
-    ls_set_ptr = p->ls_sets;
-    for (i=0 ; i < p->ls_set_am ; i++) {
-      ls_set_ptr[i].ls_nos[2] = 0;     /* initial setting */
-      for (j=0 ; j < p->dim ; j++) {
-        ls_set_ptr[i].ls_nos[j] = (int32_t) *(ptr++);
-      }
-      memset(ls_set_ptr[i].ls_mx, '\0', 9*sizeof(MYFLT));
-      /* for (j=0 ; j < 9; j++) */
-      /*   ls_set_ptr[i].ls_mx[j] = FL(0.0);  /\* initial setting *\/ */
-      for (j=0 ; j < (p->dim) * (p->dim); j++) {
-        ls_set_ptr[i].ls_mx[j] = (MYFLT) *(ptr++);
-      }
-    }
-
-    /* other initialization */
-    p->ele_vel = FL(1.0);    /* functions specific to movement */
-    if (UNLIKELY(fabs(*p->field_am) < (2+ (p->dim - 2)*2))) {
-      return csound->InitError(csound,
-                  Str("Have to have at least %d directions in vbapzmove"),
-                  2 + (p->dim - 2) * 2);
-    }
-    if (p->dim == 2)
-      p->point_change_interval = (int32_t) (CS_EKR * *p->dur
-                                        / (fabs(*p->field_am) - 1.0));
-    else if (LIKELY(p->dim == 3))
-      p->point_change_interval = (int32_t) (CS_EKR * *p->dur
-                                        / (fabs(*p->field_am) * 0.5 - 1.0));
-    else
-      return csound->InitError(csound, "%s", Str("Wrong dimension"));
-    p->point_change_counter = 0;
-    p->curr_fld = 0;
-    p->next_fld = 1;
-    p->ang_dir.azi = *p->fld[0];
-    if (p->dim == 3) {
-      p->ang_dir.ele = *p->fld[1];
-    } else {
-      p->ang_dir.ele = FL(0.0);
-    }
-    if (p->dim == 3) {
-      p->curr_fld = 1;
-      p->next_fld = 2;
-    }
-    angle_to_cart(p->ang_dir, &(p->cart_dir));
-    p->spread_base.x = p->cart_dir.y;
-    p->spread_base.y = p->cart_dir.z;
-    p->spread_base.z = -p->cart_dir.x;
-    vbap_zak_moving_control(csound,p);
-    for (i=0;i<n;i++) {
-      p->beg_gains[i] = p->updated_gains[i];
-      p->end_gains[i] = p->updated_gains[i];
-    }
-    return OK;
-}
+// code moved to vbap.c
```

Comment:
vbap.c got quite big from these merges; suppose could have kept these separate and moved into a vbap folder, but not a big issue either way.

Reply:
I preferred to keep it all together. It was hard to keep track of these opcodes.